### PR TITLE
Feature/get component by name

### PR DESCRIPTION
**Pull request by halfhorst**
Alters the functions that allow access to the underlying components in the Component Manager and makes those functions available on the interface and from the interactive simulation context.

Also unpins tables

### CODE REVIEW

File: src/vivarium/framework/components/manager.py [hunk 1]
```diff
@@ -116,27 +116,52 @@ def add_components(self, components: Union[List, Tuple]):
...
         for component in components:
             if isinstance(component, list) or isinstance(component, tuple):
                 self._add_components(component_list, component)
             else:
                 component_list.add(component)
 
-    def get_components(self, component_type: Any) -> List:
-        """Gets a list of components currently held by the component manager
-        that are instances of a specific type. Does not include managers.
+    def get_components_by_type(self, component_type: Any) -> List[Any]:
```
Comment (collijk) [inline src/vivarium/framework/components/manager.py]: component type can actually be `type` here

File: src/vivarium/framework/components/manager.py [hunk 2]
```diff
@@ -116,27 +116,52 @@ def add_components(self, components: Union[List, Tuple]):
...
+
+        Parameters
+        ----------
+        name
+            A component name.
+        Returns
+        -------
+            A component that has name ``name`` else None.
+        """
+        for c in self._components:
```
Comment (collijk) [inline src/vivarium/framework/components/manager.py]: We don't want to return ``None`` here. we want to error.

File: src/vivarium/framework/components/manager.py [hunk 3]
```diff
@@ -183,21 +208,46 @@ def add_components(self, components: Union[List, Tuple, Any]):
...
         Parameters
         ----------
         component_type
-            A component type to get, compared against internal components
+            A component type to retrieve, compared against internal components
             using isinstance().
         Returns
         -------
-            A list of components.
+            A list of components of type ``component_type``.
```
Comment (collijk) [inline src/vivarium/framework/components/manager.py]: Your return type and return docstring are incorrect.

File: src/vivarium/interface/interactive.py [hunk 4]
```diff
@@ -234,9 +234,46 @@ def get_emitter(self, event_type: str) -> Callable:
...
+            A component name.
+        Returns
+        -------
+            A component that has the name ``name`` else None.
+
+        """
+        return self.component_manager.get_component(name)
+
+    @raise_if_not_setup(system_type='component')
+    def get_components_by_type(self, component_type: Any) -> List[Any]:
```
Comment (collijk) [inline src/vivarium/interface/interactive.py]: Don't need this on the interface, I think.

File: src/vivarium/framework/components/manager.py [hunk 5]
```diff
@@ -183,21 +208,46 @@ def add_components(self, components: Union[List, Tuple, Any]):
...
+        name
+            A component name.
+        Returns
+        -------
+            A component that has name ``name`` else None.
+
+        """
+        return self._component_manager.get_component(name)
+
+    def get_components_by_type(self, component_type: Any) -> List[Any]:
```
Comment (kfwilson) [inline src/vivarium/framework/components/manager.py]: Should this be ``Union[type, Tuple[type]]`` instead of ``Any``?

Comment (halfhorst) [reply src/vivarium/framework/components/manager.py]: yup just saw that in VPH

File: src/vivarium/framework/components/manager.py (hunk 1 above)
Comment (halfhorst) [reply src/vivarium/framework/components/manager.py]: this whole time I didn't realize you meant lowercase type :man_facepalming:

**Pull request merged**